handlers/dimensions.go

```diff
@@ -308,21 +309,18 @@ func sortedTime(opts dataset.Options) dataset.Options {
 	}
 
 	output := make(map[string][]sorting)
-
 	for _, o := range opts.Items {
 		if &o.Links == nil || &o.Links.Code == nil {
 			log.Debug("options list does not contain code ids so cannot be sorted", nil)
 			break
 		}
-
 		// these codes are mmm-mmm-yyyy where the second month relates to the year
 		// e.g. `nov-jan-2014` means november 2013 - january 2014
 		// so to sort chronologically we must refer to the second month mentioned
-		code := strings.Split(o.Links.Code.ID, "-")
-		month := code[1]
-		month = strings.ToLower(month)
-		year := code[len(code)-1]
-		year = strings.ToLower(year)
+		month, year, err := splitCode(o.Links.Code.ID)
+		if err != nil {
+			log.Debug("option format is not sortable, returning flat list", log.Data{"code": o.Links.Code.ID})
```

Comment:
should it return here?  not clear how this error is acted upon, apart from being logged